packages/starknet-snap/src/rpcs/declare-contract.ts

```diff
@@ -0,0 +1,223 @@
+    await super.preExecute(params);
+  }
+
+  /**
+   * Execute the declare contract request handler.
+   * It will show a confirmation dialog to the user before signing the contract declaration.
+   *
+   * @param params - The parameters of the request.
+   * @returns the signature of the message in string array.
+   */
+  async execute(
+    params: DeclareContractParams,
+  ): Promise<DeclareContractResponse> {
+    return super.execute(params);
+  }
+
+  protected async handleRequest(
+    params: DeclareContractParams,
+  ): Promise<DeclareContractResponse> {
+    const { contractPayload, details, address } = params;
+
+    if (!(await this.getDeclareContractConsensus(params))) {
+      throw new UserRejectedOpError() as unknown as Error;
+    }
+
+    return (await declareContractUtil(
+      this.network,
+      address,
+      this.account.privateKey,
+      contractPayload,
+      details,
+    )) as DeclareContractResponse;
+  }
+
+  protected async getDeclareContractConsensus(params: DeclareContractParams) {
+    const { contractPayload, details, address } = params;
+    const components: Component[] = [];
+
+    // Add the signer address
+    components.push(
+      row(
+        'Signer Address',
+        text({
+          value: address,
+          markdown: false,
+        }),
+      ),
+    );
+
+    // Add network information
+    components.push(
+      row(
+        'Network',
+        text({
+          value: this.network.name,
+          markdown: false,
+        }),
+      ),
+    );
+
+    // Add contract details, compiled class hash, and class hash
+    if (contractPayload.contract) {
+      const contractDetails =
+        typeof contractPayload.contract === 'string'
+          ? contractPayload.contract
+          : JSON.stringify(contractPayload.contract, null, 2);
+      components.push(
+        row(
+          'Contract',
+          text({
+            value: contractDetails,
+            markdown: false,
+          }),
+        ),
+      );
+    }
+
+    if (contractPayload.compiledClassHash) {
+      components.push(
+        row(
+          'Compiled Class Hash',
+          text({
+            value: contractPayload.compiledClassHash,
+            markdown: false,
+          }),
+        ),
+      );
+    }
+
+    if (contractPayload.classHash) {
+      components.push(
+        row(
+          'Class Hash',
+          text({
+            value: contractPayload.classHash,
+            markdown: false,
+          }),
+        ),
+      );
+    }
+
+    // Add Casm details if available
+    if (contractPayload.casm) {
+      const casmDetails = JSON.stringify(contractPayload.casm, null, 2);
+      components.push(
+        row(
+          'Casm',
+          text({
+            value: casmDetails,
+            markdown: false,
+          }),
+        ),
+      );
+    }
+
+    // Add Max Fee in ETH, Nonce, and Version if they are available in details
+    if (details?.maxFee) {
```

Comment:
i just realize we didnt estimate the fee when delcare a transaction in previous code
im not sure shall we add it back, else the user will always need to pay by ETH
(we can do it in other PR)

Reply:
I think we could add it here. But then I am thinking we should maybe consider that any action that requires to pay fees (e.g. execute, and declare) should also deploy the account if not. Wdyt ?